src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceTest.java

```diff
@@ -80,6 +80,20 @@ void should_find_by_id_and_obtain_pessimistic_write_lock() {
         assertEquals(someTaskResource, actualTaskResource.get());
     }
 
+    @Test
+    void should_find_by_id_and_wait_and_obtain_pessimistic_write_lock() {
```

Comment:
There are probably more scenarios to test ... but again I think the next ticket ought to cover those.

Reply:
added a test for exception scenario